### ISSUE TITLE
Bug: [no-useless-template-literals] Missing report for `Infinity` and `NaN` 

### ISSUE THREAD

**Issue by StyleShit** [typescript-eslint/typescript-eslint#8294]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&showAST=es&fileType=.ts&code=AYEg3gkgdgZgllOAXAngX2AbgFCjAOQEN8NMg&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=&tokens=false

### Repro Code

```TypeScript
`${Infinity}`;
`${NaN}`;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-useless-template-literals": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Report on the nodes and suggest a fix

### Actual Result

There is no report

### Additional Info

Inspired by #8284 that made me curious 😅
